src/sentry/models/groupredirect.py

```diff
@@ -14,10 +24,13 @@ class GroupRedirect(Model):
     __relocation_scope__ = RelocationScope.Excluded
 
     organization_id = BoundedBigIntegerField(null=True)
-    group_id = BoundedBigIntegerField(db_index=True)
+    group = FlexibleForeignKey(
+        "sentry.Group", related_name="primary_group_of_redirect", db_constraint=False, db_index=True
+    )
```

Comment:
Let's add `db_constraint=False` here. This will let us have the convenience of the Django ORM fks, and we can figure out how to add the constraint later. Since we already have orphans in this table, adding the db level constraint would fail at the moment.

Also, since we have an FK here, no need to set `db_index=True`. It's implicitly true for all FKs